src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs

```diff
@@ -35,119 +35,120 @@
-                                 {:headers {"Content-Type" "application/json"}
-                                  :canned-response {:responseText
-                                                      (utils/->json-string (create-mock-methodconfs-import))
-                                                    :status 200
-                                                    :delay-ms (rand-int 2000)}
-                                  :method :post
-                                  :data (utils/->json-string post-data)
-                                  :on-done (fn [{:keys [success? xhr]}]
-                                             (if success?
-                                               ((:on-import props))
-                                               (js/alert (str "Error in import : "  (.-statusText xhr)))))})))}]]))})
+; Convenience method for rendering a label and text field
+(defn- formatted-label-textfield [label textfield]
+  [:div {}
+   [:div {:style {:float "left" :width "180px" :paddingTop "10px"}} label ": "] [:div {} textfield]])
 
+; Convenience method for rendering a header
+(defn- formatted-header [text]
+  [:div {}
+   [:div {:style {:fontSize 24 :align "center" :textAlign "center" :paddingBottom "0.5em"}} text]
+   [:hr]])
 
-(defn render-import-modal [config props refs on-close on-import]
-  [comps/Dialog
-   {:width 750
-    :dismiss-self on-close
-    :content (react/create-element
-               [:div {}
-                [:div {:style {:backgroundColor "#fff"
-                               :borderBottom (str "1px solid " (:line-gray style/colors))
-                               :padding "20px 48px 18px"
-                               :fontSize "137%" :fontWeight 400 :lineHeight 1}}
-                 "Import a Method Configuration"
-                 [:hr]
-                 (config "name") [:br]
-                 (config "namespace") [:br]
-                 (config "snapshotId")]
-                 [:div {:style {:position "absolute" :right 2 :top 2}}
-                  [:div {:style {:backgroundColor (:button-blue style/colors) :color "#fff"
-                                 :padding "0.5em" :cursor "pointer"}
-                         :onClick on-close}
-                   (icons/font-icon {:style {:fontSize "60%"}} :x)]]
-                [:div {:style {:padding "22px 48px 40px"
-                               :backgroundColor (:background-gray style/colors)}}
-                 [ModalImportOptionsAndButton {:init-Name (config "name")
-                                               :init-Namespace (config "namespace")
-                                               :init-SnapshotId (config "snapshotId")
-                                               :on-import on-import
-                                               :workspace-id (:workspace-id props)}]]])}])
+; Render out the import form - Need access to the selected configuration, workspace, and import callback.
```

Comment:
Rather than this comment, bind on-import in the let below so it's clear what this component needs for properties. To make it even clearer/bug-proof, add assertions too.